packages/algjulia-service/src/decapodes.jl

```diff
@@ -226,86 +297,44 @@ struct SimResult
 end
 export SimResult
 
-function SimResult(sol::ODESolution, mesh::EmbeddedDeltaDualComplex2D)
+function SimResult(sol::ODESolution, system::System)
+
+    points = collect(values(system.mesh.subparts.point.m));
 
-    points = collect(values(mesh.subparts.point.m));
+    xlen = 51; ylen = 51;
+
+    statevars = infer_state_names(system.pode)
+    statevar = length(statevars) == 1 ? first(statevars) : error("$statevars must be length one")
 
     function at_time(sol::ODESolution, timeidx::Int)
-        [SVector(i, j, sol.u[timeidx].C[51*(i-1) + j]) for i in 1:51, j in 1:51]
+        [SVector(i, j, getproperty(sol.u[timeidx], statevar)[xlen*(i-1) + j]) for i in 1:xlen, j in 1:ylen]
     end
-    # TODO indexing by "C", more principled way of indexing (not hardcoding 51)
 
     state_vals = map(1:length(sol.t)) do i
         at_time(sol, i)
     end
 
     # TODO engooden
-    SimResult(sol.t, state_vals, 0:50, 0:50)
+    SimResult(sol.t, state_vals, 0:xlen-1, 0:ylen-1)
 end
 # TODO generalize to HasDeltaSet
 
-struct System
-    pode::SummationDecapode
-    mesh::HasDeltaSet
-    dualmesh::HasDeltaSet
-    init::Any # TODO specify. Is it always ComponentVector?
+function generate(s, my_symbol; hodge=GeometricHodge())
```

Comment:
Do you still need this function? Seems like it is superseded by the close `sys_generate` above, but I'm not sure.

Reply:
No it's no longer necessary